sql/expression/convert.go

```diff
@@ -303,6 +362,22 @@ func convertValue(val interface{}, castTo string, originType sql.Type) (interfac
 	}
 }
 
+// createConvertedDecimalType creates a new Decimal type with the specified |precision| and |scale|. If a Decimal
+// type cannot be created from the values specified, an error is logged and an internal Decimal type is returned. This
+// function is intended to be used in places where an error cannot be returned (e.g. Node.Type() implementations),
+// hence why it logs an error instead of returning one.
+func createConvertedDecimalType(length, scale int) sql.DecimalType {
+	if length > 0 && scale > 0 {
+		dt, err := types.CreateColumnDecimalType(uint8(length), uint8(scale))
+		if err != nil {
+			logrus.StandardLogger().Errorf("unable to create decimal type with length %d and scale %d: %v", length, scale, err)
+			return types.InternalDecimalType
```

Comment:
Is it possible to have a `types.ErrorDecimalType` that errors on `Convert()`?
Type is called in many places, so this logs the same error a bunch of times.

Reply:
Good suggestion. I'll push a commit that caches this in the function instance the first time we create it, so that it doesn't get needlessly recreated and doesn't cause extra errors to be logged.